File: 10_Lowest_common_ancestor_in_BST.py
class Node:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None

def insert(root, key):
    if root is None:
        return Node(key)
    if key < root.key:
        root.left = insert(root.left, key)
    elif key > root.key:
        root.right = insert(root.right, key)
    return root

def LCA(root, n1, n2):
    if root is None:
        return None
    if root.key > n1 and root.key > n2:
        return LCA(root.left, n1, n2)
    if root.key < n1 and root.key < n2:
        return LCA(root.right, n1, n2)
    return root

root = None
root = insert(root, 20)
insert(root, 8)
insert(root, 22)
insert(root, 4)
insert(root, 12)
insert(root, 10)
insert(root, 14)

lca = LCA(root, 10, 14)
print(lca.key)

File: 10_Lowest_common_ancestor_in_BST.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int key;
    struct Node *left, *right;
};

struct Node* newNode(int item) {
    struct Node* temp = (struct Node*)malloc(sizeof(struct Node));
    temp->key = item;
    temp->left = temp->right = NULL;
    return temp;
}

struct Node* insert(struct Node* node, int key) {
    if (node == NULL) return newNode(key);
    if (key < node->key)
        node->left = insert(node->left, key);
    else if (key > node->key)
        node->right = insert(node->right, key);
    return node;
}

struct Node* LCA(struct Node* root, int n1, int n2) {
    if (root == NULL) return NULL;
    if (root->key > n1 && root->key > n2)
        return LCA(root->left, n1, n2);
    if (root->key < n1 && root->key < n2)
        return LCA(root->right, n1, n2);
    return root;
}

int main() {
    struct Node* root = NULL;
    root = insert(root, 20);
    insert(root, 8);
    insert(root, 22);
    insert(root, 4);
    insert(root, 12);
    insert(root, 10);
    insert(root, 14);
    struct Node* lca = LCA(root, 10, 14);
    printf("%d", lca->key);
    return 0;
}